packages/data-stores/src/reader/contexts/site-subscriptions-query-props.tsx

```diff
@@ -19,10 +19,11 @@ export const SiteSubscriptionsQueryProps = createContext< SiteSubscriptionsQuery
 	setFilterOption: () => undefined,
 } );
 
-export const SiteSubscriptionsQueryPropsProvider: React.FC< { children: React.ReactNode } > = ( {
-	children,
-} ) => {
-	const [ searchTerm, setSearchTerm ] = useState( '' );
+export const SiteSubscriptionsQueryPropsProvider: React.FC< {
+	initialSearchTerm?: string | ( () => string );
```

Comment:
Also, when `initialSearchTerm` is a function, this does state lazy init, right? I think this isn't very transparent and might need to be documented.

Reply:
I wouldn't say the init is lazy, because it happens at exactly the same time as an init with a direct `string` value. It's more precise to say that it's "expensive": we don't want to read the value from `window.location` on every render, because we need it only once, on mount.

Reply:
Ah, "lazy init" is not my term, see:

https://kentcdodds.com/blog/use-state-lazy-initialization-and-function-updates

in React docs they phrase it as "avoid recreating the initial state":

https://react.dev/reference/react/useState#avoiding-recreating-the-initial-state

We do agree that the purpose here is to keep the "expensive" work for when it must be done. This is often a reason to defer something or perform it "lazily".